Ledger-API-Test-Tool doesn't work outside the bazel environment

We use Bazel's `Runfiles` class to fix paths during test execution via Bazel. Unfortunately this breaks running tests outside the Bazel environment in the standalone ledger-api-test-tool.

The error presented to the user is the following:

```
$ java -jar ledger-api-test-tool.jar -h localhost -p 6865
Exception in thread "main" java.lang.ExceptionInInitializerError
        at com.daml.ledger.api.testtool.LedgerApiTestTool$.main(LedgerApiTestTool.scala:39)
        at com.daml.ledger.api.testtool.LedgerApiTestTool.main(LedgerApiTestTool.scala)
Caused by: java.io.IOException: Cannot find runfiles: $RUNFILES_DIR and $JAVA_RUNFILES are both unset or empty
        at com.google.devtools.build.runfiles.Runfiles.getRunfilesDir(Runfiles.java:180)
        at com.google.devtools.build.runfiles.Runfiles.create(Runfiles.java:114)
        at com.google.devtools.build.runfiles.Runfiles.create(Runfiles.java:79)
        at com.digitalasset.daml.bazeltools.BazelRunfiles.rlocation(BazelRunfiles.scala:12)
        at com.digitalasset.daml.bazeltools.BazelRunfiles.rlocation$(BazelRunfiles.scala:12)
        at com.digitalasset.daml.bazeltools.BazelRunfiles$.rlocation(BazelRunfiles.scala:16)
        at com.digitalasset.platform.PlatformApplications$Config$.<init>(PlatformApplications.scala:97)
        at com.digitalasset.platform.PlatformApplications$Config$.<clinit>(PlatformApplications.scala)
        ... 2 more
```

<!--

This issue tracker is only for technical issues related to the DAML SDK.

If you suspect that your issue has a security impact, do not use this issue tracker. In that case, please contact security@digitalasset.com and follow our responsible disclosure policy: https://digitalasset.com/security/ .

For general questions about DAML and/or support request you should use Stack Overflow: https://stackoverflow.com/questions/tagged/daml .

-->

--- a/bazel_tools/runfiles/src/main/scala/com/digitalasset/daml/bazeltools/BazelRunfiles.scala
+++ b/bazel_tools/runfiles/src/main/scala/com/digitalasset/daml/bazeltools/BazelRunfiles.scala
@@ -9,7 +9,12 @@ trait BazelRunfiles {
 
   private val MainWorkspace = "com_github_digital_asset_daml"
 
-  def rlocation(path: String): String = Runfiles.create.rlocation(MainWorkspace + "/" + path)
+  private val inBazelEnvironment =
+    Set("RUNFILES_DIR", "JAVA_RUNFILES", "RUNFILES_MANIFEST_FILE", "RUNFILES_MANIFEST_ONLY").exists(
+      sys.env.contains)
+
+  def rlocation(path: String): String =
+    if (inBazelEnvironment) Runfiles.create.rlocation(MainWorkspace + "/" + path) else path
 
 }
 

--- a/unreleased.rst
+++ b/unreleased.rst
@@ -24,3 +24,5 @@ HEAD — ongoing
 - [Ledger API] Added additional Ledger API integration tests to Ledger API Test Tool.
 - [DAML Studio] Goto definition now works on the export list of modules.
 - [Java Bindings] The artefact ``com.daml.ledger:bindings-java`` now has ``grpc-netty`` as dependency so that users don't need to explicitly add it.
+- [DAML Integration Kit] Fixed a bug in the test tool that prevented users from running the tests.
+  See `#1841 <https://github.com/digital-asset/daml/issues/1841>`__

